tests/test_iam/test_iam.py

```diff
@@ -4047,3 +4049,77 @@ def test_create_user_with_tags():
 
     resp = conn.create_user(UserName="test-create-user-no-tags")
     assert "Tags" not in resp["User"]
+
+
+@mock_iam
+def test_tag_user():
+    # given
+    client = boto3.client("iam", region_name="eu-central-1")
+    name = "test-user"
+    tags = [{"Key": "key", "Value": "value"}, {"Key": "key-2", "Value": "value-2"}]
+    client.create_user(UserName=name)
+
+    # when
+    client.tag_user(UserName=name, Tags=tags)
+
+    # then
+    response = client.list_user_tags(UserName=name)
+    response["Tags"].should.equal(tags)
```

Comment:
Looks like this fails in Py2 because of the ordering:
```
E           AssertionError: given
E           X = [{u'Value': 'value-2', u'Key': 'key-2'}, {u'Value': 'value', u'Key': 'key'}]
E               and
E           Y = [{u'Value': u'value', u'Key': u'key'}, {u'Value': u'value-2', u'Key': u'key-2'}]
E           X[0][u'Key'] is 'key-2' whereas Y[0][u'Key'] is u'key'
```

Reply:
my typical mistake with Python2, it soen't keep the order link in Python3...